### PR TITLE
[operators] Sys gen and sys search operators

### PR DESCRIPTION
**Pull request by NikitaZotov**

### CODE REVIEW

File: scp/gen_operators/SCPOperatorSysGen.cpp [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+{
+  operands = std::vector<SCPOperand*>(4);
+}
+
+std::string SCPOperatorSysGen::GetTypeName()
+{
+  return "sys_gen";
+}
+
+sc_result SCPOperatorSysGen::Parse()
```
Comment (MksmOrlov) [inline scp/gen_operators/SCPOperatorSysGen.cpp:24]: Looks like it is common method for several operators

File: scp/gen_operators/SCPOperatorSysGen.cpp [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+  {
+    if (!operands[2]->GetValue().IsValid())
+    {
+#ifdef SCP_DEBUG
+      Utils::logSCPError(ms_context, "Operand 3 has modifier FIXED, but has no value", addr);
+#endif
+      FinishExecutionWithError();
+      return SC_RESULT_ERROR_INVALID_PARAMS;
+    }
+  }
```
Comment (MksmOrlov) [inline scp/gen_operators/SCPOperatorSysGen.cpp]: Can you move all this validation code to the separate method? And reuse it in sys search for example

File: scp/gen_operators/SCPOperatorSysGen.cpp [hunk 3]
```diff
@@ -0,0 +1,270 @@
...
+    {
+#ifdef SCP_DEBUG
+      Utils::logSCPError(ms_context, "Generation parameters pair must have variable", addr);
+#endif
+      FinishExecutionWithError();
+      return SC_RESULT_ERROR_INVALID_PARAMS;
+    }
+
+    ScIterator5Ptr const replacementIt5 = ms_context->Iterator5(
+        parameterPairAddr, ScType::EdgeAccessConstPosPerm, ScType::Unknown, ScType::EdgeAccessConstPosPerm, Keynodes::rrel_2);
```
Comment (MksmOrlov) [inline scp/gen_operators/SCPOperatorSysGen.cpp]: why not basic sequence?

**Review by MksmOrlov — CHANGES_REQUESTED** [review]

**Review by ShunkevichDV — APPROVED** [review]

File: scp/gen_operators/SCPOperatorSysGen.cpp (hunk 3 above)
Comment (NikitaZotov) [reply scp/gen_operators/SCPOperatorSysGen.cpp]: Backward compatibility with old scp-machine

File: scp/gen_operators/SCPOperatorSysGen.cpp (hunk 1 above)
Comment (NikitaZotov) [reply scp/gen_operators/SCPOperatorSysGen.cpp:24]: It is true. There is basic class

File: scp/gen_operators/SCPOperatorSysGen.cpp (hunk 2 above)
Comment (NikitaZotov) [reply scp/gen_operators/SCPOperatorSysGen.cpp]: It is a problem for all operators in scp-machine

**Pull request merged**